### PR TITLE
Initial VSNetBeans extension integration tests.

### PR DESCRIPTION
**Pull request by JaroslavTulach**
Executing some VS Code extension integration tests. The execution of the [first test failed](https://travis-ci.org/github/apache/netbeans/jobs/744801153) - e.g. the travis gate correctly verifies the behavior of our extension.

Next step is to initialize the extension and get its (programmatic) version: 9219db8 - that shall succeed.

The last test creates `pom.xml` and a `.java` source in a workspace and invokes its compilation. Then it checks the result. I guess that is enough for now and quite a bit of functionality is covered.

This was failing on the gate as the "compile" command was executed sooner than language client with its `java.build.workspace` action was really initialized. This should be [fixed now](https://github.com/apache/netbeans/pull/2549/commits/3dbed9d6b1f5855ce99cc227bff928568273b722). Please review.

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
Great the [first run failed](https://travis-ci.org/github/apache/netbeans/jobs/744801153) with:
```
 [exec] npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

     [exec]     ✓ Sample test

     [exec] 

     [exec]     1) Failing test

     [exec] 

     [exec]     2) Extension test

     [exec] 

     [exec]   1 passing (39ms)

     [exec] 

     [exec]   2 failing

     [exec] 

     [exec]   1) Extension Test Suite

     [exec]        Failing test:

     [exec] 

     [exec]       AssertionError [ERR_ASSERTION]: -1 == 0

     [exec]       + expected - actual

     [exec] 

     [exec]       --1

     [exec]       +0

     [exec]       

     [exec]   	at Context.<anonymous> (out/test/suite/extension.test.js:23:16)

     [exec]   	at processImmediate (internal/timers.js:439:21)

     [exec] 

     [exec] 

     [exec]   2) Extension Test Suite

     [exec]        Extension test:

     [exec]      AssertionError [ERR_ASSERTION]: Apache NetBeans Extension is present

     [exec]   	at /home/travis/build/apache/netbeans/java/java.lsp.server/vscode/out/test/suite/extension.test.js:27:16

     [exec]   	at Generator.next (<anonymous>)

     [exec]   	at /home/travis/build/apache/netbeans/java/java.lsp.server/vscode/out/test/suite/extension.test.js:8:71

     [exec]   	at new Promise (<anonymous>)

     [exec]   	at __awaiter (out/test/suite/extension.test.js:4:12)

     [exec]   	at Context.<anonymous> (out/test/suite/extension.test.js:25:34)

     [exec]   	at processImmediate (internal/timers.js:439:21)

     [exec] 

     [exec] 

     [exec] Exit code:   1

     [exec] Done

     [exec] 

     [exec] 

     [exec] npm ERR! A complete log of this run can be found in:

     [exec] npm ERR!     /home/travis/.npm/_logs/2020-11-20T03_33_05_678Z-debug.log

BUILD FAILED
```

E.g. we can test the extension TypeScript code.

File: java/java.lsp.server/vscode/src/test/suite/extension.test.ts [hunk 1]
```diff
@@ -21,4 +28,56 @@ suite('Extension Test Suite', () => {
...
+		System.out.println("Hello World!");
+	}
+}
+		`);
+
+		vscode.workspace.saveAll();
+
+		let delayPromise = new Promise((r) => setTimeout(r, 20000));
+		vscode.window.showInformationMessage('Invoking compile command');
+		await vscode.commands.executeCommand("java.workspace.compile");
```
Comment (JaroslavTulach) [inline java/java.lsp.server/vscode/src/test/suite/extension.test.ts:64]: Can you help, @dbalek? I need to wait for the `java.workspace.compile` task to finish and just using `await` isn't enough.  Instead I have to `setTimeout(_, 20s)` and that isn't pretty at all.

Comment (JaroslavTulach) [reply java/java.lsp.server/vscode/src/test/suite/extension.test.ts:62]: f217836 might be the right answer.

**Comment by sdedic** [comment]
Great job with the client Promise !
.... but sadly I don't understand the part with Mocha in `index.ts` at all, so let's wait for a more clever reviewer...

**Review by sdedic — APPROVED** [review]

**Comment by JaroslavTulach** [comment]
> Great job with the client Promise !

Thanks. My dream is that we will be able to hide the whole state management behind a single `Promise<NbLanguageClient>` that will hold `nbProcess` and other information. But let's leave that for another PR.

Let this one focus on the _Mocha magic_ and integrate it!

**Pull request merged**